src/components/templates/Poster/usePosterVideo.ts

```diff
@@ -0,0 +1,132 @@
+import { useEffect, useMemo, useState } from "react";
+import Video, { Room, LocalParticipant, RemoteParticipant } from "twilio-video";
+import firebase from "firebase/app";
+
+import { withId, WithId } from "utils/id";
+
+import { User } from "types/User";
+
+import { useUser } from "hooks/useUser";
+import { useWorldUsersById } from "hooks/users";
+
+// TODO: create useVideoRoom hook and divide this into two hooks & refactor it
```

Comment:
> TODO: create useVideoRoom hook and divide this into two hooks & refactor it

Reply:
What's the difference between this hook and `useVideoState` from `src/components/templates/FireBarrel/useVideo.tsx`?

Any reason why we aren't using that? It already seems to handle a lot of the stuff that seems to be getting done here?

Reply:
Thanks for the idea! Moved `useVideo` from Firebarrel to hooks/ and refactored/updated it a bit in https://github.com/sparkletown/sparkle/pull/1299/commits/f11e9e2afc468857ecb95202bd490d3b2e37592b

Reply:
Sounds good. I've added a note to the TODO of https://github.com/sparkletown/sparkle/pull/1155 where this was going to be done:

![image](https://user-images.githubusercontent.com/753891/116646247-f2f3b080-a9ba-11eb-959d-f4e1f7598dbd.png)